azure-kusto-data/azure/kusto/data/request.py

```diff
@@ -516,7 +523,20 @@ def __init__(self, kcsb: Union[KustoConnectionStringBuilder, str]):
         self._auth_provider = _AadHelper(kcsb) if kcsb.aad_federated_security else None
         self._request_headers = {"Accept": "application/json", "Accept-Encoding": "gzip,deflate", "x-ms-client-version": "Kusto.Python.Client:" + VERSION}
 
+    @staticmethod
+    def _build_runtime_error(func_name: str) -> RuntimeError:

```

Comment:
I dislike build functions. these are ctors, and should be implemented as such. Also, since this is an expected error, we should probably raise it with a `Kusto` prefix like `KustoAsyncRuntimeError`

Reply:
👍